File: python/2535.difference-between-element-sum-and-digit-sum-of-an-array.py
class Solution:
    def differenceOfSum(self, nums: list[int]) -> int:
        def digitSum(n: int) -> int:
            s = 0
            while n:
                n, r = divmod(n, 10)
                s += r
            return s

        return abs(sum(nums) - sum(map(digitSum, nums)))

File: cpp/2535.difference-between-element-sum-and-digit-sum-of-an-array.cpp
#include <numeric>
#include <vector>

#include <cstdlib>

using std::vector;

class Solution {
private:
  int digitSum(int n) {
    int s = 0;
    while (n) {
      s += n % 10;
      n /= 10;
    }

    return s;
  }

public:
  int differenceOfSum(vector<int> &nums) {
    int total = std::accumulate(nums.begin(), nums.end(), 0);
    int totalDigitSum =
        accumulate(nums.begin(), nums.end(), 0,
                   [&](int a, int b) -> int { return a + digitSum(b); });
    return abs(total - totalDigitSum);
  }
};